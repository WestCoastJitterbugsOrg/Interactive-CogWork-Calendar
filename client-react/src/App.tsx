import FullCalendar from "@fullcalendar/react";
import dayGridPlugin from "@fullcalendar/daygrid";
import listPlugin from "@fullcalendar/list";
import timeGridPlugin from "@fullcalendar/timegrid";
import React from "react";
import EventContainer from "./event-selection";
import { loadCogworkData } from "./services/cogwork";
import { Wcj } from "./types";
import Button from "./shared/Button";
import './fullcalendar-custom.scss'

interface AppState {
  error: any;
  isLoaded: boolean;
  data: Wcj.EventCategory[];
}

export default class App extends React.Component {
  state: AppState = {
    error: null,
    isLoaded: false,
    data: [],
  };

  async componentDidMount() {
    try {
      const data = await loadCogworkData();
      this.setState({
        isLoaded: true,
        data,
      });
    } catch (error) {
      this.setState({
        isLoaded: true,
        error,
      });
    }
  }

  render() {
    return (
      <div className="flex flex-row">
        <div className="h-screen relative overflow-y-hidden">
          <div className="bg-wcj-black flex flex-row h-20 justify-center items-center">
            <Button title="Select all" className="mx-2"></Button>
            <Button title="Deselect all" className="mx-2"></Button>
          </div>
          <EventContainer eventCategories={this.state.data}></EventContainer>
        </div>
        <div className="flex-grow">
          <FullCalendar
            plugins={[dayGridPlugin, listPlugin, timeGridPlugin]}
            initialView="dayGridMonth"
            height='100vh'
          ></FullCalendar>
        </div>
      </div>
    );
  }
}
