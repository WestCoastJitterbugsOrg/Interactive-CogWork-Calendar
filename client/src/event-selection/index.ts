import EventGroup from "./event-group";
import EventItem from "./event";
import ToggleAllButtons from "./ToggleAllButtons";

export { EventGroup };
export { EventItem };
export { ToggleAllButtons };
