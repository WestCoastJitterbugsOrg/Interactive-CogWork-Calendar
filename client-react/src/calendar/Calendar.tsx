import FullCalendar, {
  EventInput,
  EventSourceInput,
  ToolbarInput,
} from "@fullcalendar/react";
import dayGridPlugin from "@fullcalendar/daygrid";
import listPlugin from "@fullcalendar/list";
import timeGridPlugin from "@fullcalendar/timegrid";
import { formatDate } from "@fullcalendar/common";
import "./fullcalendar-custom.css";
import { StateContext } from "../App";
import { useContext } from "react";
import { Wcj } from "../types";

type FullCalendarPropType = typeof FullCalendar.prototype.props;

const CalendarViewConfig: FullCalendarPropType["views"] = {
  dayGridMonth: {
    eventDidMount: (e) =>
      (e.el.title =
        e.event.title + "\nPlace: " + e.event.extendedProps["place"]),
    titleFormat: { year: "numeric", month: "long" },
  },
  timeGridWeek: {
    titleFormat: (args) =>
      `Week ${formatDate(args.date.marker, { week: "numeric" })}, 
      ${formatDate(args.date.marker, {
        year: "numeric",
        month: "long",
      })}`,
    dayHeaderFormat: {
      weekday: "short",
      day: "2-digit",
      month: "short"
    },
  },
  listEternal: {
    type: "list",
    titleFormat: { year: "numeric", month: "long" },
    visibleRange: (currentDate) => ({
      start: currentDate,
      end: new Date(currentDate).setFullYear(currentDate.getFullYear() + 1),
    }),
    viewDidMount: function () {
      if (this) {
        (this.headerToolbar as ToolbarInput).center = "";
      }
    },
  },
};

export default function Calendar() {
  const stateContext = useContext(StateContext);
  const events = Object.values(stateContext.state.events.byId)
    .filter((event) => event.showInCalendar)
    .map(wcj2fcEvent);
  return (
    <FullCalendar
      plugins={[dayGridPlugin, listPlugin, timeGridPlugin]}
      initialView="timeGridWeek"
      height="100vh"
      views={CalendarViewConfig}
      buttonText={{
        today: "Today",
        month: "Month",
        week: "Week",
        list: "List",
      }}
      headerToolbar={{
        start: "today,prev,next",
        center: "title",
        end: "timeGridWeek,dayGridMonth,listEternal",
      }}
      firstDay={1}
      nowIndicator
      timeZone="UTC"
      displayEventEnd
      eventTimeFormat={{
        hour: "2-digit",
        minute: "2-digit",
        meridiem: false,
        hour12: false,
      }}
      allDaySlot={false}
      eventSources={events}
    />
  );
}

function wcj2fcEvent(wcjEvent: Wcj.Event): EventSourceInput {
  return {
    id: wcjEvent.id,
    backgroundColor: wcjEvent.color,
    borderColor: wcjEvent.color,
    textColor: brightnessByColor(wcjEvent.color) > 127 ? "black" : "white",
    events: wcjEvent.occasions.map<EventInput>((occasion) => ({
      id: `${wcjEvent.id}-${occasion.start}-${occasion.end}`,
      title: wcjEvent.title,
      start: occasion.start,
      end: occasion.end,
      groupId: wcjEvent.id,
      extendedProps: wcjEvent,
    })),
  };
}

/**
 * Calculate brightness value by RGB or HEX color.
 * @param color (String) The color value in RGB or HEX (for example: #000000 || #000 || rgb(0,0,0) || rgba(0,0,0,0))
 * @returns (Number) The brightness value (dark) 0 ... 255 (light)
 */
function brightnessByColor(color: string) {
  const isHEX = color.indexOf("#") === 0;
  const isRGB = color.indexOf("rgb") === 0;

  let rgb: [number, number, number] | undefined;

  if (isHEX) {
    const m = color
      .substr(1)
      .match(color.length === 7 ? /(\S{2})/g : /(\S{1})/g);
    if (m) {
      rgb = [parseInt(m[0], 16), parseInt(m[1], 16), parseInt(m[2], 16)];
    }
  }
  if (isRGB) {
    const m = color.match(/(\d+){3}/g);
    if (m) {
      rgb = [parseInt(m[0]), parseInt(m[1]), parseInt(m[2])];
    }
  }
  if (rgb != null) {
    return (rgb[0] * 299 + rgb[1] * 587 + rgb[2] * 114) / 1000;
  } else {
    return 255;
  }
}
