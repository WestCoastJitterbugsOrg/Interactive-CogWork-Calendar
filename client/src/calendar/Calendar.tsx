import FullCalendar, {
  EventInput,
  EventSourceInput,
} from "@fullcalendar/react";
import { formatDate } from "@fullcalendar/common";
import dayGridPlugin from "@fullcalendar/daygrid";
import listPlugin from "@fullcalendar/list";
import timeGridPlugin from "@fullcalendar/timegrid";
import React from "react";
import { StateContext } from "../App";
import "./fullcalendar-custom.css";
import resolveConfig from "tailwindcss/resolveConfig";
import tailwindConfig from "../../tailwind.config.js";
import { TailwindValues } from "tailwindcss/tailwind-config";

const fullConfig = resolveConfig(tailwindConfig);

type FullCalendarPropType = typeof FullCalendar.prototype.props;

const CalendarViewConfig = (
  start: Date,
  end: Date
): FullCalendarPropType["views"] => {
  return {
    dayGridMonth: {
      eventDidMount: (e) =>
        (e.el.title =
          e.event.title + "\nPlace: " + e.event.extendedProps["place"]),
      titleFormat: { year: "numeric", month: "long" },
      dayHeaderFormat: { weekday: "long" }
    },
    timeGridWeek: {
      scrollTimeReset: false,
      scrollTime: '09:00:00',
      slotLabelFormat: {
        hour: "2-digit",
        minute: "2-digit",
        meridiem: false,
        hour12: false,
      },
      eventDidMount: (e) =>
        (e.el.title =
          e.event.title + "\nPlace: " + e.event.extendedProps["place"]),
      titleFormat: (args) =>
        `Week ${formatDate(args.date.marker, { week: "numeric" })}, 
          ${formatDate(args.date.marker, {
            year: "numeric",
            month: "long",
          })}`,
      dayHeaderFormat: (args) =>
        formatDate(args.date.marker, { weekday: "long" }) +
        "\n" +
        formatDate(args.date.marker, { day: "numeric" }),
    },
    listEternal: {
      type: "list",
      listDaySideFormat: (args) =>
        formatDate(args.date.marker, { weekday: "long" }) +
        ", Week " +
        formatDate(args.date.marker, { week: "numeric" }),
      listDayFormat: { month: "long", day: "numeric", year: "numeric" },

      buttonIcons: false,

      viewDidMount: () => {
        jQuery(
          ".fc-header-toolbar.fc-toolbar .fc-toolbar-chunk:nth-child(-n+2)"
        ).css({
          visibility: "hidden",
          width: "0",
          height: "0",
        });
      },
      viewWillUnmount: () => {
        jQuery(
          ".fc-header-toolbar.fc-toolbar .fc-toolbar-chunk:nth-child(-n+2)"
        ).css({
          visibility: "visible",
          width: "auto",
          height: "auto",
        });
      },
      eventDidMount: (e) => {
        // Add place info to events
        const title = jQuery(e.el).find(".fc-list-event-title").get(0);
        if (title != null) {
          title.outerHTML = `
            <td class="fc-list-event-title">${e.event.title}</td>
            <td class="text-right">${e.event.extendedProps["place"]}</td>
           `;
        }
        // Change colspan of day header
        const prevSibling = e.el.previousElementSibling;
        if (prevSibling?.classList.contains("fc-list-day")) {
          prevSibling
            .getElementsByTagName("th")
            .item(0)
            ?.setAttribute("colspan", "4");
        }
      },
      visibleRange: function () {
        return {
          start: start,
          end: end,
        };
      },
    },
  };
};

export default function Calendar() {
  const stateContext = React.useContext(StateContext);
  const wcjEvents = Object.values(stateContext.state.events.byId).filter(
    (event) => event.showInCalendar
  );
  const events = wcjEvents.map(wcj2fcEvent);
  const firstOccasion = Math.min(
    ...wcjEvents.flatMap((event) =>
      event.occasions.map((occ) => new Date(occ.start).getTime())
    )
  );
  const lastOccasion = Math.max(
    ...wcjEvents.flatMap((event) =>
      event.occasions.map((occ) => new Date(occ.end).getTime())
    )
  );
  return (
    <div className="wcjcal-fc">
      <FullCalendar
        plugins={[dayGridPlugin, listPlugin, timeGridPlugin]}
        initialView={
          window.innerWidth <=
          parseInt((fullConfig.theme.screens as TailwindValues)["sm"])
            ? "listEternal"
            : "timeGridWeek"
        }
        height="100%"
        views={CalendarViewConfig(
          new Date(firstOccasion),
          new Date(lastOccasion)
        )}
        buttonText={{
          today: "Today",
          month: "Month",
          week: "Week",
          list: "List",
        }}
        headerToolbar={{
          start: "today,prev,next",
          center: "title",
          end: "timeGridWeek,dayGridMonth,listEternal",
        }}
        firstDay={1}
        nowIndicator
        timeZone="UTC"
        displayEventEnd
        eventTimeFormat={{
          hour: "2-digit",
          minute: "2-digit",
          meridiem: false,
          hour12: false,
        }}
        allDaySlot={false}
        eventSources={events}
        eventBackgroundColor="#AB2814"
        eventBorderColor="#AB2814"
      />
    </div>
  );
}

function wcj2fcEvent(wcjEvent: Wcj.Event): EventSourceInput {
  return {
    id: wcjEvent.id,
    events: wcjEvent.occasions.map<EventInput>((occasion) => ({
      id: `${wcjEvent.id}-${occasion.start}-${occasion.end}`,
      title: wcjEvent.title,
      start: occasion.start,
      end: occasion.end,
      groupId: wcjEvent.id,
      extendedProps: wcjEvent,
    })),
  };
}
