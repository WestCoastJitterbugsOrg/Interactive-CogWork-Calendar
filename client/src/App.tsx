import React, { ReactChild, useMemo, useReducer, useState } from "react";
import "tailwindcss/tailwind.css";
import Calendar from "./calendar/Calendar";
import EventGroup from "./event-selection/EventGroup";
import ToggleAllButtons from "./event-selection/ToggleAllButtons";
import loadCogworkData from "./services/cogwork";
import SpinLoader from "./shared/Spinner";
import EventStore from "./store/model";
import eventReducer, { EventActions, EventActionTypes } from "./store/reducers";

const initialContext: EventStore = {
  categories: { byId: {}, allIds: [] },
  events: { byId: {}, allIds: [] },
};

export const StateContext = React.createContext<{
  state: EventStore;
  dispatch: React.Dispatch<EventActions>;
}>({
  state: initialContext,
  dispatch: () => null,
});

export default function App() {
  const [loadState, setLoadState] = useState(
    "loading" as "loading" | "loaded" | string
  );

  const [state, dispatch] = useReducer(eventReducer, initialContext);

  useMemo(async () => {
    try {
      const data = await loadCogworkData();
      setLoadState("loaded");
      dispatch({
        type: EventActionTypes.eventsLoaded,
        payload: data,
      });
    } catch (e) {
      if (e instanceof Promise) {
        e.then((val) => {
          console.log(val);
          setLoadState(val);
        });
      } else {
        setLoadState((e as any).toString());
      }
    }
  }, []);

  switch (loadState) {
    case "loading":
      return <SpinLoader />;
    case "loaded":
      return (
        <StateContext.Provider value={{ state, dispatch }}>
            <div className="flex flex-col lg:flex-row bg-white">
              <div className="max-h-screen overflow-y-hidden lg:w-96 flex flex-col">
                <div className="flex-none">
                  <ToggleAllButtons />
                </div>
                <div className="flex-grow w-full overflow-y-scroll bg-wcj-sand divide-y divide-wcj-mint">
                  {state.categories.allIds.map<ReactChild>((categoryId) => (
                    <EventGroup key={categoryId} category={categoryId} />
                  ))}
                </div>
              </div>
              <div className="flex-grow">
                <Calendar />
              </div>
            </div>
        </StateContext.Provider>
      );
    default:
      // Error message sent from server
      return (
        <div className="container m-auto my-8">
          <h1 className="text-2xl font-bold underline text-wcj-red">
            Error while loading data!
          </h1>
          <p className="font-bold">Got the following from server:</p>
          <pre className="font-mono">
            {JSON.stringify(JSON.parse(loadState), null, 4)}
          </pre>
        </div>
      );
  }
}
