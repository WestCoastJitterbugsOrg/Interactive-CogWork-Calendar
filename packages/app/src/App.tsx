import appStyle from './App.module.scss';
import { EventSeriesModal } from './EventSeriesModal';
import { Footer } from './Footer';
import { Header } from './Header';
import { Calendar } from './calendar/Calendar';
import { EventSelection } from './event-selection/EventSelection';
import { stateContext } from './state';
import { useEffect, useState } from 'react';
import { WCJ } from './types';
import { canStoreSelection } from './services/cookies';

type Props = WCJ.Context & {
	colors: Record<string, string>;
	parent: HTMLElement;
	isLoading: boolean;
};

export function App(props: Props) {
	useEffect(() => {
		for (const color in props.colors) {
			const colorVal = props.colors[color];
			props.parent.style.setProperty(`--cw-color-${color}`, colorVal);
		}
	}, [props.colors, props.parent]);

	const [eventModal, setEventModal] = useState<string>();
	const [checkedEvents, setCheckedEvents] = useCheckedEvents(
		props.events.map((x) => x.id),
		props.selectedEventIds,
	);

	return (
		<stateContext.Provider
			value={{
				categories: props.categories,
				events: props.events,
				checkedEvents,
				setCheckedEvents,
				eventModal,
				setEventModal,
			}}
		>
			<Header />
			<div className={appStyle.contentWrapper}>
				<div className={appStyle.eventSelection} data-testid="event-selection">
					<EventSelection isLoading={props.isLoading} />
				</div>
				<div className={appStyle.calendar} data-testid="calendar">
					<Calendar events={props.events} checkedEvents={checkedEvents} />
				</div>
			</div>
			<Footer />
			<EventSeriesModal
				parent={props.parent}
				event={props.events.find((event) => event.id === eventModal)}
				close={() => {
					setEventModal(undefined);
				}}
			/>
		</stateContext.Provider>
	);
}

export function useCheckedEvents(
	allEventIds: string[],
	initialSelection: string[],
) {
	// This is a hook to store the user selection events into localstorage to keep it over multiple sessions.
	// Since new events can be added from one session to another, and we want the user to be able to see the new event,
	// we actually store the unselected events rather than the selected ones.

	const [checkedEvents, setCheckedEvents] =
		useState<string[]>(initialSelection);

	// When checkedEvents is changed, put it in localStorage
	useEffect(() => {
		if (canStoreSelection()) {
			const uncheckedEvents = allEventIds.filter(
				(eventId) => !checkedEvents.includes(eventId),
			);
			localStorage.setItem('uncheckedEvents', JSON.stringify(uncheckedEvents));
		}
	}, [allEventIds, checkedEvents]);

	return [checkedEvents, setCheckedEvents] as const;
}
