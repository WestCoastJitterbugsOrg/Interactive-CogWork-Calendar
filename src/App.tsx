import Calendar from "./calendar";
import EventSelection from "./event-selection/EventSelection";
import initContext from "./services/cogwork";
import { EventSeriesModal } from "./shared";
import StateWrapper from "./store/StateWrapper";

export default function App() {
  return (
    <div className="flex flex-row flex-wrap items-stretch bg-white min-h-[calc(100vh-32px)]">
      <StateWrapper initialContext={initContext()}>
        <div className="flex flex-col flex-grow w-96 max-h-[calc(100vh-32px)]">
          <EventSelection />
        </div>
        <div
          className="flex-grow flex-shrink-0 min-h-[calc(100vh-32px)] min-w-[calc(100%-384px)]"
          data-testid="calendar-wrapper"
        >
          <Calendar />
        </div>
        <EventSeriesModal />
      </StateWrapper>
    </div>
  );
}
