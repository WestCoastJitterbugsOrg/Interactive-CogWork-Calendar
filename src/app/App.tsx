import appStyle from './App.module.scss';
import { EventSeriesModal } from './EventSeriesModal';
import { Footer } from './Footer';
import { Header } from './Header';
import { Calendar } from './calendar/Calendar';
import { EventSelection } from './event-selection/EventSelection';
import { StateWrapper } from './store/StateWrapper';
import { useEffect, useState } from 'react';
import { WCJ } from 'types';

type Props = WCJ.Context & { colors: Record<string, string> };

export function App(props: Props) {
	const [rootRef, setRef] = useState<HTMLElement | null>(null);
	useEffect(() => {
		for (const color in props.colors) {
			const colorVal = props.colors[color];
			rootRef?.style.setProperty(`--cw-color-${color}`, colorVal);
		}
	}, [props.colors, rootRef]);

	return (
		<>
			<Header />
			<StateWrapper categories={props.categories} events={props.events}>
				<div
					id="cw-filter-calendar-root"
					className={appStyle.root}
					ref={setRef}
				>
					<div className={appStyle.contentWrapper}>
						<aside className={appStyle.eventSelection}>
							<EventSelection />
						</aside>
						<main className={appStyle.calendar} data-testid="calendar-wrapper">
							<Calendar />
						</main>
					</div>
					<Footer />
				</div>
			</StateWrapper>
			{rootRef && <EventSeriesModal parent={rootRef} />}
		</>
	);
}
