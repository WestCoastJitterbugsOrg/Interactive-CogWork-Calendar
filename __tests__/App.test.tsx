import Calendar from "@app/calendar";
import { EventGroup } from "@app/event-selection";
import { createPopper } from "@popperjs/core";
import "@testing-library/jest-dom";
import { render as libRender, RenderResult, act } from "@testing-library/react";
import { ReactElement } from "react";
import { unmountComponentAtNode } from "react-dom";
import App, { StateContext } from "../src/App";
import { defaultEventData } from "../__mocks__/cwEvents";
import { mockStateContext } from "../__mocks__/stateContext";

type Global = typeof globalThis & {
  wcjcal_ajax_obj?: typeof defaultEventData;
};
jest.mock("@popperjs/core", () => {
  const originalModule = jest.requireActual("@popperjs/core");
  return {
    __esModule: true,
    ...originalModule,
    createPopper: jest.fn(() => "mocked createPopper"),
  };
});


function render(ui: ReactElement) {
  if (container != null) {
    return libRender(ui, { container });
  } else {
    throw "Container not initialized";
  }
}

function getEventGroupInputs(
  renderResult: RenderResult,
  groupSelector = (collection: HTMLCollection) => collection[0]
) {
  const eventGroup = groupSelector(
    renderResult.getByTestId("event-selection-groups").children
  ).children;

  const group = eventGroup[0];
  const eventsWrapper = eventGroup[1];

  const groupInput = group.querySelector("input");
  const eventInputs = Array.from(
    eventsWrapper.children[0].querySelectorAll("input")
  );

  return [groupInput, eventInputs] as const;
}

let container: HTMLDivElement | null = null;

beforeEach(() => {
  (global as Global)["wcjcal_ajax_obj"] = defaultEventData;

  container = document.createElement("div");
  container.id = "wcjcal";
  document.body.appendChild(container);
});

afterEach(() => {
  if (container != null) {
    unmountComponentAtNode(container);
    container.remove();
  }
  container = null;
  delete (global as Global)["wcjcal_ajax_obj"];
});

it("Snapshot", async () => {
  render(
    <StateContext.Provider value={mockStateContext}>
      <EventGroup category="Lindy Hop" />
    </StateContext.Provider>
  );
  expect(container?.innerHTML).toMatchSnapshot();
});

it("Events are selected at start", async () => {
  const renderResult = render(<App />);

  const [groupInput, eventInputs] = getEventGroupInputs(renderResult);
  expect(groupInput).toBeChecked();
  eventInputs.forEach((eventInput) => expect(eventInput).toBeChecked());
});

it("Clicking deselect all deselects all events and groups", async () => {
  const renderResult = render(<App />);
  const deselectAllBtn = await renderResult.findByText("Deselect all");

  act(() => deselectAllBtn.click());

  const [groupInput, eventInputs] = getEventGroupInputs(renderResult);
  expect(groupInput).not.toBeChecked();
  eventInputs.forEach((eventInput) => expect(eventInput).not.toBeChecked());
});

it("Click event in calendar opens popper", async () => {
  jest.useFakeTimers();
  jest.spyOn(global, "setTimeout");


  const initialDate =
    mockStateContext.state.events.byId["1"].occasions[0].start;


  const renderResult = render(
    <StateContext.Provider value={mockStateContext}>
      <Calendar initialDate={initialDate} />
    </StateContext.Provider>
  );

  const fcEventContainer = renderResult.container.querySelector(
    ".fc-event-title-container"
  ) as HTMLDivElement;

  act(() => {
    fcEventContainer.click();
    jest.runAllTimers();
  });

  // renderResult.debug();
  expect(createPopper).toHaveBeenCalledTimes(1);
});

