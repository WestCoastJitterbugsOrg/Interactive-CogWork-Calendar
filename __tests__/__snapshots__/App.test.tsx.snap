// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
"<div class=\\"bg-wcj-sand\\"><div class=\\"bg-wcj-cyan text-white font-bold min-h-8 py-2 flex flex-row items-center\\"><div class=\\"cursor-pointer flex-grow flex items-center\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 mx-2 transition duration-200 transform \\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\"><path stroke-linecap=\\"square\\" stroke-width=\\"4\\" d=\\"M12 4v16m8-8H4\\"></path></svg>Lindy Hop</div><div class=\\"mx-3 flex\\"><input type=\\"checkbox\\" class=\\"appearance-none w-5 h-5 bg-wcj-sand border-wcj-black border-solid border-[2px] rounded-md
                  after:content-[''] after:relative after:block after:opacity-0

                  indeterminate:after:opacity-100 indeterminate:after:top-[6px] indeterminate:after:left-[2px] indeterminate:after:w-3 indeterminate:after:h-1 
                  indeterminate:after:bg-wcj-black indeterminate:after:rounded-1 indeterminate:after:rotate-0

                  checked:after:opacity-100 checked:after:left-[6px] checked:after:-top-[5px] checked:after:w-2 checked:after:h-4 checked:after:rotate-[50deg] 
                  checked:after:border-solid checked:after:border-0
                  checked:after:border-b-[3px] checked:after:border-r-[3px] checked:after:border-wcj-black checked:after:rounded-br-[3px]
                  \\" readonly=\\"\\" checked=\\"\\"></div></div><div class=\\"bg-wcj-sand overflow-hidden max-h-0\\"><div class=\\"flex flex-row items-center px-2 my-2 min-h-8\\"><div class=\\"h-4 w-4 mr-2 flex-none\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"block text-black opacity-50 hover:opacity-100 cursor-pointer\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\"><path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z\\"></path></svg></div><div class=\\"flex-grow cursor-pointer pr-2 leading-tight\\">Grundkurs Lindy Hop</div><div class=\\"flex flex-none cursor-pointer\\"><input type=\\"checkbox\\" class=\\"
        appearance-none w-6 h-6 border rounded-full border-opacity-50 border-solid border-black

        after:content-[''] after:relative after:block after:opacity-0 
        after:left-[8px] after:top-[3px] after:w-[6px] after:h-[12px]
        after:rotate-45 after:border-solid after:border-white after:border-0
        
        checked:after:border-b-[2px] checked:after:border-r-[2px] checked:after:opacity-100
      \\" style=\\"background-color: rgb(171, 40, 20);\\" readonly=\\"\\" checked=\\"\\"></div></div></div></div>"
`;
